File: Linkbilding/Google_tables_URL_chistim/google_tables_chistim_URL.py
import re

def extract_urls_from_file(input_path):
    with open(input_path, 'r', encoding='utf-8') as file:
        text = file.read()

    # Разделяем текст на потенциальные URL
    potential_urls = re.split(r'\s+', text)

    # Отфильтровываем потенциальные URL, оставляя только валидные
    valid_urls = [url for url in potential_urls if re.match(r'https?://\S+', url)]

    return valid_urls

def save_to_file(urls, output_path):
    with open(output_path, 'w', encoding='utf-8') as file:
        for url in urls:
            file.write(url + "\n")

if __name__ == "__main__":
    input_path = r"Linkbilding\Google_tables_URL_chistim\input\source.txt"
    output_path = r"Linkbilding\Google_tables_URL_chistim\output\urls.txt"

    urls = extract_urls_from_file(input_path)
    save_to_file(urls, output_path)

File: Linkbilding/Google_tables_URL_chistim/google_tables_chistim_URL.go
package main

import (
	"bufio"
	"os"
	"regexp"
	"strings"
)

func extractUrlsFromFile(inputPath string) []string {
	file, err := os.Open(inputPath)
	if err != nil {
		panic(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	text := ""
	for scanner.Scan() {
		text += scanner.Text() + " "
	}

	potentialUrls := strings.Fields(text)
	urlPattern := regexp.MustCompile(`https?://\S+`)
	validUrls := []string{}

	for _, url := range potentialUrls {
		if urlPattern.MatchString(url) {
			validUrls = append(validUrls, url)
		}
	}

	return validUrls
}

func saveToFile(urls []string, outputPath string) {
	file, err := os.Create(outputPath)
	if err != nil {
		panic(err)
	}
	defer file.Close()

	writer := bufio.NewWriter(file)
	for _, url := range urls {
		writer.WriteString(url + "\n")
	}
	writer.Flush()
}

func main() {
	inputPath := "Linkbilding/Google_tables_URL_chistim/input/source.txt"
	outputPath := "Linkbilding/Google_tables_URL_chistim/output/urls.txt"

	urls := extractUrlsFromFile(inputPath)
	saveToFile(urls, outputPath)
}
